### DISCUSSION TITLE
Results extraction script

### DISCUSSION THREAD

**Discussion by motomatters** [fabricepipart/mgp-timings#20] [Show and tell]
Hi Fabrice, 

Last night I hacked together a script to extract the results from the MotoGP JSON endpoint and format them into the HTML table I use on the website. It's written in Perl and based in no small part on Len's script. I wrote it in Perl because it's been a long time since I did any work in Java.

It still needs a lot of work. I've added in the most immediate needs in the TODO at the top of the file. And it's sloppy and ugly, but it works. 

It's attached to this. As it's not Java, I haven't added it to the current code base (also because I haven't used GitHub before, it wasn't a thing when I stopped coding professionally). So it's up to you what you want to do with it. At the very least, it will give you an idea of what I doing with the data.
[motogp-json-results.tar.gz](https://github.com/fabricepipart/mgp-timings/files/8305159/motogp-json-results.tar.gz)

**Comment by fabricepipart** [comment]
Thanks a lot!
I give a look at your scripts ASAP and I'll try to provide the same level of functionality right after

**Comment by fabricepipart** [comment]
Thanks for the script. That was helpful. I am almost there now. I can combine the info from the riders API, the results API and the PDF content. It needs a bit of polishing and I'll release it.

It still requires a bit of work since the race and practice sessions PDFs / API content are different. Plus I faced the funny bug... Exceptionnally in 2021 Maverick Vinales has been in two teams (just to make my life a bit more complex!).

Just a question though: do you want me to create an endpoint that returns the html content? A whole page? A snippet? Or you dont care and csv or json is just fine?

**Reply by fabricepipart** [reply]
Here is the equivalent of what you proposed. Here you can download a session as CSV:

https://mgp-timings.teknichrono.fr/session/2021/FRA/motogp/rac/results/details/csv
